library/components/TCsvExport.vue

```diff
@@ -33,25 +33,25 @@ export default {
       type: String,
       default: "Download CSV",
     },
+    additionalFilters:{
+      type: Object,
+      default: ()=>{},
+    },
   },
   data: () => ({
     is_loading: false,
   }),
   methods: {
     download() {
       this.is_loading = true;
-      const queryString = window.location.search;
-      const urlParams = new URLSearchParams(queryString);
-      let filters = {};
-      let filenameFilters = "";
-      for (let filterKey of urlParams.keys()) {
-        filters[filterKey] = urlParams.get(filterKey);
-        filenameFilters += "-" + filterKey + "_" + urlParams.get(filterKey);
-      }
       payload = {
-        filters,
         layer: this.tLayer,
       };
+      filters = this.getFiltersState ? this.getFiltersState : {}
```

Comment:
We might change the default/initial value for `filters` in the page store to be `{}` instead of `null`. Here: https://github.com/snyk/topcoat-core/blob/main/frontend/src/store/modules/page.js#L40

Reply:
Is that something that I can just update now as part of this PR?

Reply:
Oh no, I did not mean to change it right now, it was a thought while watching you have to do this validation. We can change that on a separated task.

Reply:
Awesome